pkg/agent/prommetrics.go

```diff
@@ -15,53 +15,76 @@ type PromMetrics struct {
 	runnerRestarts            prometheus.Counter
 }
 
+func register[P prometheus.Collector](reg *prometheus.Registry, collector P) P {
+	reg.MustRegister(collector)
+	return collector
+}
+
 func makePrometheusParts(globalstate *agentState) (PromMetrics, *prometheus.Registry) {
-	schedulerRequests := prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "autoscaling_agent_scheduler_plugin_requests_total",
-			Help: "Number of attempted HTTP requests to the scheduler plugin by autoscaler-agents",
-		},
-		[]string{"code"},
-	)
-	informantRequestsOutbound := prometheus.NewCounterVec(
+	var metrics PromMetrics
+	reg := prometheus.NewRegistry()
+
+	// register stock collectors directly:
+	//   (even though MustRegister is variadic, the function calls
+	//   are cheap and calling it more than once means that when
+	//   it panics, we know exactly which metric caused the error.)
+	reg.MustRegister(collectors.NewGoCollector())
+	reg.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
+
+	// the register() function registers the collector and returns
+	// it so we can set it directly on the output structure.
+	metrics.schedulerRequests = register(reg,
+		prometheus.NewCounterVec(
```

Comment:
```suggestion
	metrics.schedulerRequests = register(reg, prometheus.NewCounterVec(
```

so that formatting matches with the ones below